### ISSUE TITLE
Example doesn't work, and question about module filters

### ISSUE THREAD

**Issue by JP-Ellis** [cardoe/stderrlog-rs#7]
I'm not entirely sure why, but the example provided does not work (tested with the clap example).  In particular, the issue seems to be with the module filtering: 
```rust
    stderrlog::new()
        // .module(module_path!())   // <-- this
        .verbosity(verbose)
        .init()
        .unwrap();
```

I'm using Rust `1.23`, with Clap `2.29.1`, log `0.4.1` and `0.3.9`, and stderrlog `0.2.3`.

It doesn't help that there is no documentation as to what is exactly meant by "module path" and how it works.  Do I have to really specify *all* modules I want to allow the logs, or are sub-modules allowed by default?  What if I want to exclude a particular sub-module?

For example, suppose the executable is called `my_exec` and has the structure:

```
main.rs
foo/mod.rs
foo/bar.rs
foo/baz.rs
hello.rs
```

Is the path to `foo` just `foo`, or `my_exec::foo` or `::foo`?

What if I want to log messages from a particular library?

**Comment by JP-Ellis** [comment]
Great thanks!  What about the entry-point to the executable?  So in the example above, how would I allow messages from `main.rs`?  Is it just `main`?

**Comment by cardoe** [comment]
Should just be the name of the crate or if a multi binary crate then the binary. I’ll have to double check when I am not on mobile but I’ll be improving the docs soon.

**Comment by ndarilek** [comment]
I'm also experiencing this. Threw me for a loop for quite some time until I found this issue. https://gitlab.com/ndarilek/tma Uncomment the `module!` definition and logging fails. Even an `error!` call immediately after the logger initializes shows nothing.

**Comment by cardoe** [comment]
While I know this one has hung around for a bit and I'm sorry for that I've struggled to reproduce it.

```
$ cargo run --example clap -- -vvvvv
warning: unused manifest key: package.changelog
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/debug/examples/clap -vvvvv`
TRACE - trace message
DEBUG - debug message
INFO - info message
WARN - warn message
ERROR - error message
```

**Comment by cardoe** [comment]
> Great thanks! What about the entry-point to the executable? So in the example above, how would I allow messages from main.rs? Is it just main?

It should be `main`.

**Comment by cardoe** [comment]
> I'm also experiencing this. Threw me for a loop for quite some time until I found this issue. https://gitlab.com/ndarilek/tma Uncomment the module! definition and logging fails. Even an error! call immediately after the logger initializes shows nothing.

I can reproduce it with your project so I'll fix it up using that as a base.

**Comment by cardoe** [comment]
@ndarilek It looks like the issue you're running into is specific to `log` 0.4.x and I'm going to release `stderrlog` 0.4.0 to address that.

@JP-Ellis Were you perhaps using `log` 0.4.x in your application as well?

**Comment by cardoe** [comment]
@JP-Ellis See the `large-example` added under `examples/`. It should hopefully answer all the questions you have.

**Comment by cardoe** [comment]
@ndarilek & @JP-Ellis the issue you had with `log` 0.4.x is fixed as of `stderrlog` 0.3.0 as well. So both issues should be solved.

I've further improved the documentation post 0.3.0 so its not rendering in docs.rs yet but feel free to ask some more questions if the docs aren't clear enough.

**Comment by spl** [comment]
I'm brand new to Rust and, of course, to `stderrlog`. I ran into the same problem as people above when trying the example code in the documentation – all three of which include `.module(module_path!())` – because I had structured my source into separate modules. It took me quite a bit of time before I realized why my logging was not appearing.

Since (a) an example is intended to demonstrate simple, basic usage, (b) module-level logging is arguably a slightly more advanced use case that you'd look for after seeing too many messages, and (c) there have been several reports of confusion here, I think it would be better to remove `.module(...)` from the documentation examples (e.g. [here](https://docs.rs/stderrlog/0.4.1/stderrlog/#simple-use-case)).

Alternatively, if you prefer to keep `.module(...)` in the examples, I would request that a comment be placed next to it that emphasizes that this restricts logging to only that module. That would hopefully help people like me who don't even realize a thing called “module-level logging” exists and therefore don't search through the documentation to find it.
